lib/modem_jwt/modem_jwt.c

```diff
@@ -148,6 +159,76 @@ int modem_jwt_generate(struct jwt_data *const jwt)
 	return ret;
 }
 
+int copy_uuid_str(char *dst, const char *src, const char *field)
+{
+	/* nRF modem generated tokens have two fixed fields that contain what we are looking for
+	 * "iss":"<chip>.<UUID>"  This is device UUID
+	 * "jti":"<chip>.<UUID>.<JTI data>" This is FW UUID
+	 * So I need to seek a first dot after the correct tag is found.
+	 */
+	char *tag = strstr(src, field);
+	if (!tag) {
+		return EINVAL;
+	}
+	tag += strlen(field);
+	char *end = strchr(tag, '"');
+	char *dot = strchr(tag, '.');
+	if (!end || !dot || dot > end) {
+		return EINVAL;
+	}
+	char *uuid = dot + 1;
+	size_t len = end - uuid;
+	if (len < NRF_UUID_V4_STR_LEN) {
+		return EINVAL;
+	}
+	strncpy(dst, uuid, NRF_UUID_V4_STR_LEN);
```

Comment:
you could change this to: `strncpy(dst, uuid, NRF_UUID_V4_STR_LEN + 1);`
then delete the following line.

Reply:
Actually no.
`strncpy()` does not add NUL termination if the source is longer than `count` parameter.
https://en.cppreference.com/w/c/string/byte/strncpy

In this case, we are copying substring from a longer JSON string, so I need to terminate it.

`strlcpy()` would work like you suggest, but that is BSD extension. It is part of Newlib, but requires `__BSD_VISIBLE` flag, and I have not seen anyone using that, so I'll stay C99 compatible.

Reply:
ah yes, `uuid` isn’t null terminated here 👍